lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -1924,22 +2110,33 @@ LogicalResult IfOp::verify() {
   return success();
 }
 
-/// Returns the last EnableOp within the child tree of 'parentSeqOp'. If no
-/// EnableOp was found (e.g. a "calyx.par" operation is present), returns
-/// None.
-static std::optional<EnableOp> getLastEnableOp(SeqOp parent) {
+/// Returns the last EnableOp within the child tree of 'parentSeqOp' or
+/// `parentStaticSeqOp.` If no EnableOp was found (e.g. a "calyx.par" operation
+/// is present), returns None.
+template <typename OpTy>
+static std::optional<EnableOp> getLastEnableOp(OpTy parent) {
+  static_assert(std::is_same<SeqOp, OpTy>() ||
```

Comment:
Suggestion:

```
namespace {

// A struct to enforce that the LHS template is one of the RHS templates.
// For example:
//   std::is_any<uint32_t, uint16_t, float, int32_t>::value is false.
template <class T, class... Ts>
struct is_any : std::disjunction<std::is_same<T, Ts>...> {};

}
```

With C++17, this should work. Then you can do:

```
is_any<OpTy, SeqOp, StaticSeqOp>::value
```

This should simplify a lot of these logical ORs.